deeplay/components/rnn.py

```diff
@@ -1,147 +1,128 @@
-    - rnn_type (str): Type of RNN to use ('LSTM', 'GRU') (Default: 'LSTM')
-
-    Constraints
-    -----------
-    - input shape: (batch_size, seq_len, input_size)
-    - output shape: (batch_size, seq_len, hidden_size * num_directions)
-
-    Evaluation
-    ----------
-    >>> output, hidden = self.rnn(x)
-    >>> return output, hidden
-
-    Examples
-    --------
-    >>> # With default configuration
-    >>> rnn = BaseRNN(Config().update(input_size=my_input_size, hidden_size=my_hidden_size))
-    >>> # With custom number of layers and dropout
-    >>> rnn = BaseRNN(Config().update(input_size=my_input_size, hidden_size=my_hidden_size, num_layers=2, dropout=0.1, rnn_type='GRU'))
-
-    Return Values
-    -------------
-    The forward method returns a tuple of two elements:
-    - outputs: the output features `h` from the last layer of the RNN, for each timestep
-    - hidden: the hidden state for the last timestep for each layer, `h_n`
-
-    Additional Notes
-    ----------------
-    The `Config` class is used for configuring the RNN. For more details refer to [Config Documentation](#).
-    """
-
-    @staticmethod
-    def defaults():
-        return (
-            Config()
-            .input_size(None)  # Expected to be set by user
-            .hidden_size(256)
-            .num_layers(1)
-            .dropout(0.1)
-            .rnn_type('LSTM')  # Can be 'LSTM' or 'GRU'
-        )
-
-    def __init__(self, config):
-        super().__init__()
-        self.config = config
-        self.input_size = config.get('input_size')
-        self.hidden_size = config.get('hidden_size', 256)
-        self.num_layers = config.get('num_layers', 1)
-        self.dropout = config.get('dropout', 0)
-        self.rnn_type = config.get('rnn_type', 'LSTM')
-
-        rnn_class = nn.LSTM if self.rnn_type == 'LSTM' else nn.GRU
-        self.rnn = rnn_class(
-            self.input_size,
-            self.hidden_size,
-            self.num_layers,
-            dropout=self.dropout,
-        )
-
-    def forward(self, x, lengths=None):
-        if lengths is not None:
-            x = pack_padded_sequence(x, lengths)
-        outputs, hidden = self.rnn(x)
-        if lengths is not None:
-            outputs, _ = pad_packed_sequence(outputs)
-        return outputs, hidden
-
-class EncoderRNN(BaseRNN):
-    """Encoder RNN module.
-
-    This module extends the BaseRNN module to create an RNN encoder with GRU units.
-
-    Configurables
-    -------------
-    Inherits all configurables from BaseRNN, with the addition of:
-    - bidirectional (bool): If `True`, becomes a bidirectional RNN (Default: True)
-    - Embedding (nn.Embedding): Embedding layer for the input (Default: None)
+
+class Drop(nn.Module):
+    def __init__(self, p=0):
+        super(Drop, self).__init__()
+        self.p = p
+
+    def forward(self, x):
+        return nn.Dropout(p=self.p)(x[0]),x[1]
```

Comment:
Use nn.functional.dropout if possible.

Reply:
Why is this preferred? NN.Dropout automatically disables dropout in model evaluation mode, for instance

Reply:
This implementation will not be disablet in eval mode, because it's not attached to the parent module. In that case, one should create the module in the init method